### PR TITLE
feat: resource name

### PR DESCRIPTION
**Pull request by nic-chen**
### Summary
add `name` field for route, service, upstream to name each instance, and it is  unique, can't be duplicate.

### CODE REVIEW

File: t/admin/upstream.t [hunk 1]
```diff
@@ -1146,7 +1146,7 @@ GET /t
 
 
 
-=== TEST 35: type chash, hash_on: consumer, don't need upstream key
+=== TEST 35: type chash, hash_on: consumer, donot need upstream key
```
Comment (membphis) [inline t/admin/upstream.t:1149]: `donot` -> `do not`

**Comment by moonming** [comment]
@nic-chen Is this PR reday for review?

**Comment by nic-chen** [comment]
yes , please.  @moonming

**Comment by membphis** [comment]
@nic-chen missing doc

**Comment by nic-chen** [comment]
done. @membphis

**Pull request merged**